cmd/oras/internal/option/target.go

```diff
@@ -102,14 +102,13 @@ func (opts *Target) Parse() error {
 		if len(opts.headerFlags) != 0 {
 			return errors.New("custom header flags cannot be used on an OCI image layout target")
 		}
-		return nil
+		_, _, err := opts.parseOCILayoutReference()
```

Comment:
Is it possible to save the path and reference here and remove the `parseOCILayoutReference` call in `newXXX` here?

```suggestion
		opts.Path, opts.Reference, err := opts.parseOCILayoutReference()
```

Reply:
saved the values here and removed the references of `parseOCILayoutReference`